plonky2x/core/src/frontend/hash/curta/mod.rs

```diff
@@ -131,14 +156,20 @@ pub trait SHA<L: PlonkParameters<D>, const D: usize, const CYCLE_LEN: usize>:
                 // The digest bit is equal to zero for all chunks except the one that corresponds to
                 // the `chunk_index`. We find the bits by comparing each value between 0 and the
                 // total number of chunks to the `chunk_index`.
-                let mut flag = builder.constant::<BoolVariable>(true);
+                let mut t_var = builder.constant::<U32Variable>(0);
+                let chunk_size = builder.constant::<U32Variable>(128);
                 for j in 0..total_number_of_chunks {
+                    if HAS_T_VALUES {
+                        t_var = builder.add(t_var, chunk_size);
+                    }
                     let j_var = builder.constant::<U32Variable>(j as u32);
-                    let lte = builder.lte(last_chunk_index, j_var);
-                    let lte_times_flag = builder.and(lte, flag);
```

Comment:
can you explain what's going on here? why can you just remove these lines?

Reply:
I refactored the original logic (which I think is now easier to read).  So just to give some context, this loop is calculating the digest_bits (an array of bits to label which message chunk is the last chunk of the message) and t_values arrays.  Those arrays will be part of the curta public input.

The original logic is similar to:

```
digest_bits = []
flag = true
for j in range(0, num_chunks):
    if j <= last_chunk_idx AND flag:
        digest_bits[j] = false
    if j > last_chunk_idx:
        flag = false
```

I changed it to:
```
digest_bits = []
t_values = []
t_val = 0
for j in range(0, num_chunks):
    at_digest_bit = (j == last_chunk_idx)
    digest_bits[j] = at_digest_bit
 
   t_val += chunk_size
   if at_digest_bit:
      t_val = msg_length
   
   t_values[j] = t_val
```